src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -272,21 +265,10 @@ impl HummockIterator for ConcatSstableIterator {
                     ord == Ordering::Less || ord == Ordering::Equal
                 })
                 .saturating_sub(1); // considering the boundary of 0
-            let mut next_idx = table_idx;
-            if VersionedComparator::compare_key(
-                &self.tables[table_idx].key_range.as_ref().unwrap().right,
-                key,
-            ) == Ordering::Less
-            {
-                next_idx += 1;
-                self.sstable_iter.take();
-            }
-            while next_idx < self.tables.len() {
-                self.seek_idx(next_idx, Some(key)).await?;
-                if self.sstable_iter.is_some() {
-                    break;
-                }
-                next_idx += 1;
+            self.sstable_iter.take();
```

Comment:
`seek_idx` function will take the sstable_iter

Reply:
fixed.